mirror_builder/wheels.py

```diff
@@ -8,11 +8,13 @@
 
 
 def build_wheel(ctx, req, sdist_root_dir, build_env):
-    logger.info('building wheel for %s in %s', req.name, sdist_root_dir)
+    logger.info('building wheel for %s in %s writing to %s', req.name, sdist_root_dir,
+                ctx.wheels_build)
     builder = overrides.find_override_method(req.name, 'build_wheel')
     if not builder:
         builder = _default_build_wheel
-    return builder(ctx, build_env, req, sdist_root_dir)
+    builder(ctx, build_env, req, sdist_root_dir)
+    return ctx.wheels_build.glob('*.whl')
```

Comment:
The API for build_wheel() is that it returns the name(s) of the wheel files created. The overrides used to need to produce the list, because that code was the only thing that knew where the files were. As part of this change, those files are always expected to be in the `ctx.wheels_build` directory, so it's not necessary for the override functions to deal with that any more.

Reply:
Well how does this glob work? The command shutil.move command has just moved all these wheels out of this directory to the wheels_download directory. Did I understand this right?

Reply:
There are 2 separate paths through this code.

In "bootstrap mode", driven by `handle_requirement()` in sdists.py, `build_wheel()` is called before `update_wheel_mirror()` and the return value of `build_wheel()` is not used. https://github.com/dhellmann/rebuilding-the-wheel/blob/write-to-wheels-repo/mirror_builder/sdist.py#L54-L57

In "step by step mode", driven by separate calls in `__main__.py`, the list of wheel files is printed out, but they aren't being moved into the downloads directory because the thing that builds the wheels doesn't run a wheel server. https://github.com/dhellmann/rebuilding-the-wheel/blob/write-to-wheels-repo/mirror_builder/__main__.py#L105-L113

The return value is only needed in that step-by-step mode, and only then to report on what was done.

Reply:
Makes sense. Thanks for explaining.